### PR TITLE
[Asset Processor] Document for the skipping startup scan mode

### PR DESCRIPTION
**Pull request by junbo75**
## Change summary
A new feature is added to the Asset Processor recently to skip startup scan, which helps to improve Editor's launch time when no assets are changing. Check https://github.com/o3de/o3de/pull/10056 for more details.

This PR add document for it. 

I updated the weight values for some pages to make sure that the "Skip Startup Scan" page is next to the "Faster Scanning" page, since both of them are settings for the Asset Processor.

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [ ] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/assets/asset-processor/skip-startup-scan.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+linkTitle: Skip Startup Scan
+title: Asset Processor Skip Startup Scan
+description: Skips startup checks for assets modified while Asset Processor was closed.
+weight: 700
+toc: true
+---
+
+Skipping Startup Scan improves O3DE Editor's launch time if no assets are modified while Asset Processor was closed. You can switch startup scan modes at any time but need to restart Asset Processor for the change to take effects. Asset Processor saves your preference between sessions.
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/asset-processor/skip-startup-scan.md:9]: Maybe talk to the why, ie use this mode for code development workflows or use it only when you are sure you are not making asset changes or asset processing impacting changes. Its really an "expert" mode to aid code updating and debugging iteration times.

Comment (amzn-mike) [inline content/docs/user-guide/assets/asset-processor/skip-startup-scan.md:9]:
```suggestion
Skipping Startup Scan improves O3DE Editor's launch time if no assets are modified while Asset Processor was closed. You can switch startup scan modes at any time but need to restart Asset Processor for the change to take effect. Asset Processor saves your preferences between sessions.

```

File: content/docs/user-guide/assets/asset-processor/skip-startup-scan.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+description: Skips startup checks for assets modified while Asset Processor was closed.
+weight: 700
+toc: true
+---
+
+Skipping Startup Scan improves O3DE Editor's launch time if no assets are modified while Asset Processor was closed. You can switch startup scan modes at any time but need to restart Asset Processor for the change to take effects. Asset Processor saves your preference between sessions.
+
+When Skip Startup Scan is enabled:
+1. Asset Processor skips startup checks on asset modifications after launch and enters the idle state directly.
+2. O3DE Editor receives an message from Asset Processor which indicates that all the assets are ready to use.
```
Comment (amzn-mike) [inline content/docs/user-guide/assets/asset-processor/skip-startup-scan.md:13]:
```suggestion
2. O3DE Editor receives a message from Asset Processor which indicates that all the assets are ready to use.
```

**Review by amzn-mike — APPROVED** [review]

File: content/docs/user-guide/assets/asset-processor/skip-startup-scan.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+---
+
+Skipping Startup Scan improves O3DE Editor's launch time if no assets are modified while Asset Processor was closed. You can switch startup scan modes at any time but need to restart Asset Processor for the change to take effect. Asset Processor saves your preferences between sessions.
+
+Skipping Startup Scan is an expert mode that is designed for aiding code development workflows (e.g. code updating and debugging). Use it only when you are not modifying the asset processing pipeline or making asset content changes.
+
+When Skip Startup Scan is enabled:
+1. Asset Processor skips startup checks on asset modifications after launch and enters the idle state directly.
+2. O3DE Editor receives a message from Asset Processor which indicates that all the assets are ready to use.
+3. O3DE Editor is unblocked to launch.
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/asset-processor/skip-startup-scan.md:16]: nit: Maybe add a line that AP will begin to watch for asset changes once it goes to Idle state?

**Review by lmbr-pip — APPROVED** [review]

File: content/docs/user-guide/assets/asset-processor/skip-startup-scan.md (hunk 3 above)
Comment (junbo75) [reply content/docs/user-guide/assets/asset-processor/skip-startup-scan.md:16]: Just added the line.

File: content/docs/user-guide/assets/asset-processor/skip-startup-scan.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+linkTitle: Skip Startup Scan
+title: Asset Processor Skip Startup Scan
+description: Skips startup checks for assets modified while Asset Processor was closed.
+weight: 700
+toc: true
+---
+
+Skipping Startup Scan improves O3DE Editor's launch time if no assets are modified while Asset Processor was closed. You can switch startup scan modes at any time but need to restart Asset Processor for the change to take effect. Asset Processor saves your preferences between sessions.
+
+Skipping Startup Scan is an expert mode that is designed for aiding code development workflows (e.g. code updating and debugging). Use it only when you are not modifying the asset processing pipeline or making asset content changes. Asset Processor will begin to watch for asset changes once it goes to the Idle state.
```
Comment (micronAMZN) [inline content/docs/user-guide/assets/asset-processor/skip-startup-scan.md:11]: Some suggestions for style.

```suggestion
Skipping Startup Scan is intended to be an aid for rapid code iteration workflows (edit, build, and debug cycles, for example). Use it only when you aren't modifying the asset processing pipeline or making changes to source assets. Asset Processor watches for asset changes once it enters the Idle state.
```

**Review by micronAMZN — CHANGES_REQUESTED** [review]
Most of the suggestions are to adjust the topic weights so that the topics appear in the same order in the ToC as the index page.

A comment on the UX implementation because I see this code is merged: I think the "enable-to-disable" paradigm will confuse some users. When a toggle switch is active, users should expect it to *do this thing* (enable startup scan), when disabled, *don't do this thing* (disable startup scan).

**Comment by junbo75** [comment]
> Most of the suggestions are to adjust the topic weights so that the topics appear in the same order in the ToC as the index page.
> 
> A comment on the UX implementation because I see this code is merged: I think the "enable-to-disable" paradigm will confuse some users. When a toggle switch is active, users should expect it to _do this thing_ (enable startup scan), when disabled, _don't do this thing_ (disable startup scan).

Yeah, @rainbj also has some concerns around the descriptions for this mode in the AP GUI. We have an internal ticket for tracking the UX/Document requirements and will make further updates after this initial check-in.

**Review by micronAMZN — APPROVED** [review]

**Comment by rainbj** [comment]
@junbo75 can you tag me in the final text. Thanks!

**Pull request merged**